### PR TITLE
docs: add contribution guidelines

### PR DESCRIPTION
**Pull request by Xazin**
First step in having contribution guidelines.

Closes: #76

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+Before filing a feature request, try searching in the issues for keywords related to the feature request you have in mind, there is a chance that someone else already filed the same or a similar request!
+# <a name="branches"></a> Branches naming conventions
+We use simple naming conventions for branches, to easily navigate and track multiple branches across features. Branches follow a 3 part logic, the first part describes the type of work is being done on the branch.
+
+<a name="tags"></a>Often used tags are: `feat`, `feature`, `chore`, `task`, `refac`, `refactor`, `fix`, `bugfix`, `docs`, `documentation`, etc.
+
+The second part is what issue or feature the branch relates to. This is used by taking the issue number, and prefixing it with either `gh-` or `can-`.
+
+Examples: `gh-97`, `can-104`, etc.
```
Comment (rubenhorn) [inline docs/CONTRIBUTING.md:31]: Maybe explicitly distinguish between referencing GitHub issues and Jira tasks.

**Comment by yasht01** [comment]
Should we mention our Slack (invite link?) in this document?

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+<tag>(<scope>): <subject>
+<newline>
+<body>
+<newline>
+<footer>
+```
+### Tag
+The tag describes the type of work that was actually done in the commit. This can differentiate from the branch name which the commit comes from, in case something changed and the purpose of the commit is not the same as it was intended. Most of the times, the commit tag will be the same as the branch tag.
+
+See [example tags](#tags) from the branch naming conventions section.
```
Comment (rubenhorn) [inline docs/CONTRIBUTING.md:58]: Example tags missing?

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+Don't use underscore (`_`) in branch naming, and don't use uppercase letters.
+# <a name="commits"></a> Commit conventions
+To ensure our commit history stays clean, and can more easily be used for tracking, we follow some simple conventions that go hand in hand with our [branches naming conventions](#branches).
+
+The format for the commit title is:
+ `<tag>(<scope>): <subject>`
+
+The format for a commit with body and footer is:
+```
```
Comment (rubenhorn) [inline docs/CONTRIBUTING.md:48]: Maybe also add a complete example of a commit message (like example branch names above)

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (yasht01) [reply docs/CONTRIBUTING.md:58]: It redirects to line 27.

**Comment by Xazin** [comment]
> Should we mention our Slack (invite link?) in this document?

Nah, slack is our internal tool. Maybe in the future we can have an external communication channel, but Slack should always stay private and internal, we discuss information that is not open source in public channels.

In the future maybe a discord or similar.

**Review by rubenhorn — DISMISSED** [review]

**Review by rubenhorn — APPROVED** [review]

**Review by yasht01 — APPROVED** [review]

**Review by wizlif — APPROVED** [review]

**Pull request merged**